src/header/TransferBench.hpp

```diff
@@ -2088,6 +2272,1114 @@ namespace {
     return ERR_NONE;
   }
 
+// IB Verbs-related functions
+//========================================================================================
+
+#ifndef NO_IBV_EXEC
+#define MAX_SEND_WR_PER_QP 12
+#define MAX_RECV_WR_PER_QP 12
+const unsigned int rdmaFlags = IBV_ACCESS_LOCAL_WRITE    |
+                               IBV_ACCESS_REMOTE_READ    |
+                               IBV_ACCESS_REMOTE_WRITE   |
+                               IBV_ACCESS_REMOTE_ATOMIC;
+#define IB_PSN  0
+#define INIT_ONCE(flag)\
+  do {                 \
+  if(flag) return;     \
+  else flag = true;    \
+  } while(0);
+
+const  uint64_t WR_ID = 1789;
+static ibv_device** deviceList = nullptr;
+static int RdmaNicCount        = -1;
+static std::vector<std::string> IbDeviceBusIds;
+static std::vector<std::set<int>> NicToGpuMapper;
+static std::vector<int> GpuToNicMapper;
+static std::vector<std::string> DeviceNames;
+static int GpuCount;
+static bool DeviceMappingInit  = false;
+static bool PcieTreeInit = false;
+static bool MultiportFlag = false;
```

Comment:
naming convention is UpperCamelCase for functions, lowerCamelCase for variables.
We may need to be careful here as these variables are still going to be accessible by anyone including the header file.  Perhaps add underscore prefix, or take a second look to see if these are all necessary to be "global".

I think the better approach should have a accessor function that captures this with static variables within the function instead.

e.g.:
ErrResult GetIbvDevice(int devIndex, ibv_device*& ibvDevice) {
  static ibv_device** deviceList = nullptr;
  if (deviceList == nullptr) {
   // initialize
   }
   // Error check for valid devIndex
   ...
}

Reply:
Thanks for the suggested workaround for globals. I think this can be its own PR. I have temporarily used and underscore prefix and adjusted the naming convention

Reply:
Resolved by merging the recent PR from @gilbertlee-amd's branch